applications/portal/backend/api/mappers/antibody_mapper.py

```diff
@@ -9,22 +10,26 @@
 from api.utilities.exceptions import AntibodyDataException
 from cloudharness import log
 from openapi.models import Antibody as AntibodyDTO
+from openapi.models import AbstractAntibody as AbstractAntibodyDTO
 from .mapping_utils import dict_to_snake, dict_to_camel, to_snake
 from ..services.gene_service import get_or_create_gene
 from ..services.specie_service import get_or_create_specie
-from ..utilities.functions import extract_base_url
 
 dto_fields = {to_snake(f) for f in AntibodyDTO.__fields__}
 
 
-class AntibodyMapper(IDAOMapper):
+def extract_base_url(url):
+    return urlsplit(url).hostname
+
 
-    def from_dto(self, dto: AntibodyDTO) -> Antibody:
+class AntibodyMapper(IDAOMapper):
 
+    def from_dto(self, dto: AbstractAntibodyDTO) -> Antibody:
         if hasattr(dto, "abId") and dto.abId:
-            ab: Antibody = Antibody.objects.get(dto.abId)
+            ab: Antibody = Antibody.objects.get(ab_id=dto.abId)
         else:
             ab = Antibody()
+            ab.ab_id = 0
```

Comment:
@filippomc I don't know why we need this. I moved it to the only place I think it could make some sense but I'm not sure